Automattic/woocommerce-services
# Add API support for creating packages with POST and updating packages with PUT

## Description


Endpoint proposal: p91TBi-4xT-p2

Currently, there is one endpoint `POST /connect/packages` to **update** all the packages. In Woo mobile, we'd like to **create** packages without modifying the existing packages - it prevents the client from accidentally deleting packages that it hasn't fetched since mobile might not be always in sync with the backend.

This PR added support for creating packages by updating the existing `POST /connect/packages` that updates all packages to `PUT /connect/packages` and implementing `POST /connect/packages` to create new packages. Below is a list of major changes:

#### API
- Added functions to create custom and predefined packages in `WC_Connect_Service_Settings_Store`
- In `WC_REST_Connect_Base_Controller`, added support to handle PUT requests for a controller
- In `WC_REST_Connect_Packages_Controller`:
  - Updated the existing POST handling to create packages
  - Updated the previous POST handling that updates packages to PUT

#### Client
- Added support for PUT requests
- Updated existing packages update request from `post` to `put`

### Testing steps



Please confidence check on the existing behavior to update packages in wp-admin:

#### WCS settings

- Go to the WCS settings under `WooCommerce --> Settings --> Shipping --> WooCommerce Settings` (`/wp-admin/admin.php?page=wc-settings&tab=shipping&section=woocommerce-services-settings`)
- In the Packaging section, tap "Add package"
- Add a custom package or select from any of the service packages
- Tap "Add package" to add the package(s)
- Tap "Save changes" at the bottom of the section --> the success banner "Your shipping settings have been saved." should appear, and the new package(s) should be shown after reloading the settings page

#### Order details > create shipping label

- Go to an order where a shipping label can be created
- Tap "Create shipping label" CTA
- Expand the Packaging tab, and tap "Add package" next to "Package details"
- Add a custom package or select from any of the service packages
- Tap "Add package" to add the package(s) --> the success banner "Your shipping packages have been saved.", and the created packages should appear in the "Package details" dropdown list

### Checklist
